forte/data/readers/imdb_reader.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The reader that reads IMDB data into data pack.
+Data Overview:
+https://www.kaggle.com/lakshmi25npathi/imdb-dataset-of-50k-movie-reviews
+Data Format:
+https://www.kaggle.com/lakshmi25npathi/imdb-dataset-of-50k-movie-reviews
+"""
+import logging
+import os
+from typing import Iterator
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.data_pack import DataPack
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Sentence, Token, Document
+
+
+__all__ = [
+    "IMDBReader"
+]
+
+
+class IMDBReader(PackReader):
+    r""":class:`IMDBReader` is designed to read
+        in the imdb review dataset used
+        by sentiment classification task.
+        The Original data format:
+        "movie comment, positive"
+        "movie comment, negative"
+    """
+
+    def _collect(self, *args, **kwargs) -> Iterator[str]:
+        r"""Iterator over text files in the data_source
+
+        Args:
+            args: args[0] is the directory to the .imdb files.
+            kwargs:
+
+        Returns: Iterator over files in the path with imdb extensions.
+        """
+
+        imdb_directory: str = args[0]
+
+        imdb_file_extension = "imdb"
+
+        logging.info(type(kwargs))
+
+        logging.info("Reading dataset from %s with extension %s",
+                     imdb_directory, imdb_file_extension)
+        return dataset_path_iterator(imdb_directory, imdb_file_extension)
+
+    def _cache_key_function(self, imdb_file: str) -> str:
+        return os.path.basename(imdb_file)
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        pack: DataPack = DataPack()
+        text: str = ""
+        offset: int = 0
+
+        with open(file_path, "r", encoding="utf8") as f:
+            for line in f:
+                line = line.strip()
+                if line != "":
+                    line_list = line.split("\",")
+                    sentence = line_list[0].strip("\"")
+                    sentiment = line_list[1]
+
+                    # Add sentence.
+                    senobj = Sentence(pack, offset + 1,
+                                      offset + len(sentence) + 1)
+                    senobj.speaker = sentiment
```

Comment:
Why are we using speaker to store the sentiment? Please use the sentiment tag.

Reply:
I think I should leave some comment here. This was a workaround for storing the tag at that time.

1. Solution1: Add a new field and pass that filed as parameter.
There is one sentiment tag under Sentence, but it is a `Dict`, which seems not suitable for storing one `str`. (Originally, Ziqian writes something like `senobj.sentiment[sent] = sentiment`. So one simple way to solve this problem is that adding another field in `Sentence` and use it.

2. Solution2: Use function as parameter but there is restriction. 
However, we might want to make the extractor more flexible to get the attribute in the `Sentence`. For example, passing a function, which can do something like "fn(Sentence) -> Attribute". Then this problem can be solved. But we come across a problem, this function cannot be serialized, unless this function is a member function of some class inside the Forte. In that case, when we deserialize the function, we can find it by its reference. (Because, when we do serialization on function, we are only storing its reference instead of the real definition. There is library that can serialize the definition though).
So to sum up, if we want to pass a function, we probably need to define that function, maybe, under the Ontology.

Reply:
Why do you want to serialize the function?

Reply:
Because we want to store the extractor and use use it for the prediction. And this function is one field of it.
If we use lambda function, it will throw some error. Let me ask Ziqian to try putting a non lambda function.

Reply:
Fixed this problem by writing a new extractor.